spec/scenarios/charge_models/prorated_graduated_spec.rb

```diff
@@ -0,0 +1,399 @@
+            plan:,
+            properties: {
+              graduated_ranges: [
+                {
+                  from_value: 0,
+                  to_value: 5,
+                  per_unit_amount: '10',
+                  flat_amount: '100',
+                },
+                {
+                  from_value: 6,
+                  to_value: 15,
+                  per_unit_amount: '5',
+                  flat_amount: '50',
+                },
+                {
+                  from_value: 16,
+                  to_value: nil,
+                  per_unit_amount: '2',
+                  flat_amount: '0',
+                },
+              ],
+            },
+          )
+
+          travel_to(DateTime.new(2023, 9, 10)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '2' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 16)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '5' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 20)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '-6' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 25)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '10' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 26)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '4' },
+              },
+            )
+          end
+
+          travel_to(DateTime.new(2023, 9, 30)) do
+            create_event(
+              {
+                code: billable_metric.code,
+                transaction_id: SecureRandom.uuid,
+                external_customer_id: customer.external_id,
+                properties: { amount: '60' },
+              },
+            )
+
+            fetch_current_usage(customer:)
+            expect(json[:customer_usage][:amount_cents].round(2)).to eq(19033)
+            expect(json[:customer_usage][:total_amount_cents].round(2)).to eq(19033)
```

Comment:
Style/NumericLiterals: Use underscores(_) as thousands separator and separate every 3 digits with them. (https://rubystyle.guide#underscores-in-numerics)